module-4/backend/week_two.md

```diff
@@ -5,18 +5,43 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+  
 3. What are the four pillars of Object Oriented programming?
+  Polymophrism- that in object oriented programming, objects can be instantiated into multiple different forms, while retaining shared behavior, yet still individualizing itself if desired. 
+  Inheritance- that classes can inherit functions/methods, and thus behavior, of other 'higher' classes and fellow object creators. 
+  Encapsulation- the phenomenon of an object being a case of internal methods/properties/attributes, and can be 'moved around' and still retain its same functionality and properties despite changing locations and environments. 
+  Abstraction- keeping the more complex code buried 'deeper' and further away from the user, refactored into organized and structured sub- or sister- files. 
+  
 4. What are some tools available in JavaScript to help you write object oriented code?
+  Hashes serve as encapsulators for JS Objects, and can encapsulate both properties and functions (attributes and methods)
+  
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+  In javascript functions must be exported, which requires something like Webpack to share and require functions.
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+  Callback functions are functions that are executed by other functions, as arguments. We use them in order to create a chain of actions and so we can refactor out code. 
+  
 7. What's the scope of variables in Javascript?
+  Var/Const variables (in Es5/Es6 respectively) are automatically hoisted to the beginning of their highest scope, while let (Es6) is a block scoped variable, which means its contained to its scope. This can mess with your code, so gotta be careful.
+
 8. What's the difference between `==` and `===` in JavaScript?
+  == is strictly equal, which functions more or less like == in Ruby, as an operator to determine likeness in code. For instance 2 == 2 would evaluate to true, while '1' == 'abc' would return false. === is an identity operator and evaluate to true only if the two objects/variables are identical, meaning they share object ids as well. 
+  
 9. Why do front end frameworks exist?
+  To create an interactive experience for the users, to carry data, and to process and display dynamic information after brower loads. 
 
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+  I really don't know.
```

Comment:
if you don't know a question here, move on but then come back afterwards and do some research and fill it in!